.github/workflows/test_workflow_scripts/golang-docker.sh

```diff
@@ -80,10 +80,27 @@ for i in {1..2}; do
 
     echo "Recorded test case and mocks for iteration ${i}"
 done
+sleep 10
 
-# Start the keploy in test mode.
+# Start keploy in test mode.
 test_container="ginApp_test"
-sudo -E env PATH=$PATH ./../../keployv2 test -c 'docker run -p8080:8080 --net keploy-network --name ginApp_test gin-mongo' --containerName "$test_container" --apiTimeout 60 --delay 20 --generate-github-actions=false &> "${test_container}.txt"
+sudo -E env PATH=$PATH ./../../keployv2 test -c "docker run -p8080:8080 --rm --name $test_container --network keploy-network gin-mongo" --containerName "$test_container" --apiTimeout 30 --delay 30 --generate-github-actions=false &> "${test_container}.txt"
+if grep "ERROR" "${test_container}.txt"; then
+    echo "Error found in pipeline..."
+    cat "${test_container}.txt"
+    exit 1
+fi
+# Monitor Docker logs for race conditions during testing.
+if grep "WARNING: DATA RACE" "${test_container}.txt"; then
+    echo "Race condition detected in test, stopping pipeline..."
+    cat "${test_container}.txt"
+    exit 1
+fi
+
+sleep 10
+
+rerecord_container="ginApp_rerecord"
+sudo -E env PATH=$PATH ./../../keployv2 rerecord -c "docker run -p8000:8000 --rm --name $rerecord_container --network keploy-network gin-mongo" --containerName "$rerecord_container" --inCi=true &> "${rerecord_container}.txt"
 
 if grep "ERROR" "${test_container}.txt"; then
```

Comment:
we're writing logs into rerecord_container.txt but checking test_container.txt for error or race, needs to be fixed.